### PR TITLE
Refactor to use new CourseContent model

### PR DESCRIPTION
**Pull request by rshipp**
Task: https://app.asana.com/0/1143269577537246/1148998666960365/f

Rename `Content` to `CourseContent`, and add some new columns to track metadata.

### CODE REVIEW

**Comment by rshipp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/course_contents_controller.rb  2
         

Clones added
============
- app/controllers/course_contents_controller.rb  3
- spec/controllers/course_contents_controller_spec.rb  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/platform/platform/pullRequest?prid=4502901&bid=15178975)

File: db/schema.rb [hunk 1]
```diff
@@ -43,13 +43,16 @@
...
-  create_table "contents", force: :cascade do |t|
+  create_table "course_contents", force: :cascade do |t|
     t.string "title"
     t.text "body"
     t.datetime "published_at"
     t.string "content_type"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
+    t.integer "course_id"
+    t.string "secondary_id"
```
Comment (sadleb) [inline db/schema.rb]: what is secondary id?

Comment (rshipp) [reply db/schema.rb]: @sadleb Depends on the `content_type` - for a `wiki_page` it'd be the `page_id`.

Comment (StetsenTech) [reply db/schema.rb]: @rshipp If it is variable...then maybe it is worth adding a comment about what it could potential refer to.

Comment (rshipp) [reply db/schema.rb]: @StetsenTech where's a good place to comment on that? in `schema.rb`? in the db as a comment on the column?

Comment (StetsenTech) [reply db/schema.rb]: @rshipp I can't think of a better place to put it, so `schema.rb` should be fine.

**Pull request merged**